### ISSUE TITLE
end-to-end wav2vec 2.0 model is not possible?

### ISSUE THREAD

**Issue by YooSungHyun** [facebookresearch/fairseq#4445]
### Before asking:
- [x] 1. search the issues.
- [x] 2. search the docs.

#### What is your question?
I want to training end-to-end style
(wav2vec 2.0 -> ctc) full training once

i think, i have enough data, we don`t need to learn pre-training.
in some specific domain, i think audio+text end-to-end learning is good for find sound feature to text!
(noisy condition like Construction site)

that is just my hyposis, i don`t experiment yet.
how about others?

but maybe, "make end-to-end style" means,
1) can not use gumble softmax
2) can not use perplexity loss
that same looks like, ESPNet architecture...

am i right?

--> It just academic question!
